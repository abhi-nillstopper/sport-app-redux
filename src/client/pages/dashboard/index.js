import React, { useEffect, useState, useMemo, useContext } from "react";
import {
  Image,
  Button,
  ButtonGroup,
  Alert,
  SplitButton,
  Dropdown,
  Container,
} from "react-bootstrap";
import moment from "moment";
import { io as socketio } from "socket.io-client";
import { useSelector, useDispatch } from "react-redux";
import "./dashboard.css";
import api from "../../service/api";
import { ReactComponent as DeletIcon } from "../../assets/dustbin.svg";
import { fetchEvents } from "../../actions/dashboard_action";
import { logoutHandler } from "../../actions/authentication_action";

//dashboard will shoow all the events
export default function Dashboard({ history }) {
  const dispatch = useDispatch();
  const radios = [
    { name: "All Sport", value: "" },
    { name: "My Events", value: "myevents" },
    { name: "Running", value: "running" },
    { name: "Cycling", value: "cycling" },
    { name: "Swimming", value: "swimming" },
  ];

  const [isLoading, SetIsLoading] = useState(true);
  const [radioValue, setRadioValue] = useState("");
  const [error, setError] = useState(false);
  const [errorMessage, setErrorMessage] = useState("");
  const [success, setSuccess] = useState(false);
  const [successMessage, setSuccessMessage] = useState("");
  const [registartionRequests, setRegistartionRequests] = useState([]);

  const user_id = localStorage.getItem("user_id");
  // const user = localStorage.getItem("user");
  const isLoggedIn = useSelector((state) => {
    return state.authentication.isLoggedIn;
  });

  const events = useSelector((state) => {
    return state.dashboard.events;
  });

  useEffect(() => {
    isLoggedIn && history.push("/");
    !isLoggedIn && history.push("/login");
  }, [isLoggedIn]);

  const getEvents = async (filter) => {
    try {
      await dispatch(fetchEvents(filter));
      SetIsLoading(false);
    } catch (error) {
      console.log(error);
      errorHandler(error.message, true);
    }
  };

  useEffect(() => {
    getEvents(radioValue);
  }, [radioValue]);

  useEffect(() => {
    if (!isLoading) {
      //disable event dropdown
      document.querySelector(
        `div[name="select-event-dropdown"]`
      ).children[0].disabled = true;
    }
  }, [isLoading]);

  const socket = useMemo(
    () =>
      socketio(process.env.REACT_APP_AXIOS_BASE_URL.trim(), {
        query: { user_id },
      }),
    [user_id]
  );

  useEffect(() => {
    socket.on("registration_request", (resp) => {
      setRegistartionRequests([...registartionRequests, resp]);
    });
  }, [registartionRequests, socket]);


  const dropDownHandler = (eventKey = "", event = "") => {
    setRadioValue(eventKey);

    document.querySelector(
      `div[name="select-event-dropdown"]`
    ).children[0].innerHTML = event.target.innerHTML;
  };

  const errorHandler = (message, logout = false) => {
    setError(true);
    setErrorMessage(message);
    setTimeout(() => {
      setError(false);
      logout ? userLogout() : dropDownHandler();
    }, 2000);
  };

  const userLogout = async () => {
    localStorage.removeItem("user_id");
    localStorage.removeItem("user");
    await dispatch(logoutHandler());
    history.push("/login");
  };

  const successHandler = (message) => {
    setSuccess(true);
    setSuccessMessage(message);
    setTimeout(() => {
      setSuccess(false);
      getEvents(radioValue);
    }, 2000);
  };

  const handleDelete = async (event_id) => {
    try {
      await api.delete(`/express/event/${event_id}`);
      successHandler("Event deleted successfully");
    } catch (error) {
      errorHandler(error.response.data.message);
    }
  };

  const handleParticipate = async (event) => {
    try {
      await api.post(`/express/registration/${event._id}`, {});
      successHandler(`Participation requested for ${event.title}`);
    } catch (error) {
      errorHandler(error.response.data.message);
    }
  };

  const handleApprove = async (eventId) => {
    try {
      await api.post(`/express/registration/${eventId}/approval`, {});
      successHandler(`Request Approved for ${eventId}`);
      removeNotification(eventId);
    } catch (error) {
      errorHandler(error);
    }
  };
  const handleReject = async (eventId) => {
    try {
      await api.post(`/express/registration/${eventId}/rejection`, {});
      successHandler(`Request Rejected for ${eventId}`);
      removeNotification(eventId);
    } catch (error) {
      errorHandler(error);
    }
  };

  const removeNotification = (eventId) => {
    const newEvents = registartionRequests.filter(
      (request) => request._id !== eventId
    );
    setRegistartionRequests(newEvents);
  };

  return (
    <Container className="dashboard-container">
      {!isLoading ? (
        <>
          <ul className="notification">
            {registartionRequests.map((request) => {
              return (
                <li key={request._id}>
                  <span>
                    <strong>{request.user.email}</strong> is requesting to
                    participate in your event&nbsp;
                    <strong>{request.event.title}</strong> &nbsp;
                  </span>
                  <ButtonGroup>
                    <Button
                      variant="primary"
                      onClick={() => handleApprove(request._id)}
                    >
                      Approve
                    </Button>
                    <Button
                      variant="danger"
                      onClick={() => handleReject(request._id)}
                    >
                      Reject
                    </Button>
                  </ButtonGroup>
                </li>
              );
            })}
          </ul>

          {success && (
            <Alert variant="success" className="event-validation">
              {successMessage}
            </Alert>
          )}

          {error && (
            <Alert variant="danger" className="event-validation">
              {errorMessage}
            </Alert>
          )}

          <div className="filter-panel">
            <SplitButton
              variant={"info"}
              name="select-event-dropdown"
              title={"Sport"}
            >
              {radios.map((radio, idx) => (
                <Dropdown.Item
                  key={radio.value}
                  eventKey={radio.value}
                  value={radio.value}
                  active={radio.value === radioValue}
                  onSelect={dropDownHandler}
                >
                  {radio.name}
                </Dropdown.Item>
              ))}
            </SplitButton>
          </div>
          <ul className="events-list">
            {events.map((event) => {
              return (
                <li key={event._id}>
                  <header>
                    {event.user === user_id && (
                      <div>
                        <DeletIcon onClick={() => handleDelete(event._id)} />
                      </div>
                    )}
                    <Image src={event.thumbnail_url}></Image>
                  </header>
                  <strong>{event.title}</strong>
                  <span>
                    Event Date: {moment(event.date).format("DD,MMM,YY")}
                  </span>
                  <span>Event Sport: {event.sport}</span>
                  <span>
                    Event Price: (in â‚¹) {parseFloat(event.price).toFixed(2)}
                  </span>
                  <span>Event Description: {event.description}</span>
                  <Button
                    variant="primary"
                    onClick={() => handleParticipate(event)}
                  >
                    Participate
                  </Button>
                </li>
              );
            })}
          </ul>
        </>
      ) : (
        <></>
      )}
    </Container>
  );
}
