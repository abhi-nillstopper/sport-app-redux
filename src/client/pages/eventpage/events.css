.new-event-container{
  width: fit-content;
  background-color: #f2f2f2;
  padding: 1.5em;
  border-radius: 1em;
}

label#thumbnail {
  margin-bottom: 20px;
  border: 1px;
  background-size: contain;
  background-repeat: no-repeat;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 120px;
}

label#thumbnail div input[type="file"] {
  display: none;
}

label#thumbnail.has-thumbnail {
  border: 0;
}

label#thumbnail.has-thumbnail img {
  display: none;
}

.event-validation{
    margin-top: 12px;
}

.new-event-container svg{
  width: 15vh;
  height: 15vh;
}
