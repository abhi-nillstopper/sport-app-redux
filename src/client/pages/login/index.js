import React, { useState, useContext, useEffect } from "react";
import { Form, Button, Container, Alert } from "react-bootstrap";
import api from "../../service/api";
import { UserContext } from "../../user-context";
import "./login.css";

export default function Login({ history }) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState(false);
  const [errorMessage, setErrorMessage] = useState("");
  const { isLoggedIn, setIsLoggedIn } = useContext(UserContext);

  useEffect(() => {
    isLoggedIn && history.push("/");
  }, []);

  const handleSubmit = async (e) => {
    e.preventDefault();
    const response = await api.post("/express/login", { email, password });
    const user_id = response.data.user_id || false;
    const user = response.data.user || false;

    try {
      if (user_id) {
        localStorage.setItem("user_id", user_id);
        localStorage.setItem("user", user);
        setIsLoggedIn(true);
        history.push("/");
      } else {
        const { message } = response.data;
        errorHandler(message);
      }
    } catch (err) {
      errorHandler("Error, server returned the error");
    }
  };

  const errorHandler = (message) => {
    setError(true);
    setErrorMessage(message);
    setTimeout(() => setError(false), 2000);
  };

  return (
    <Container className="login-container">
      <h2>Login:</h2>
      <p>
        Please <strong>login</strong> to your account
      </p>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId="formBasicEmail">
          <Form.Label>Email address</Form.Label>
          <Form.Control
            type="email"
            name="email"
            placeholder="Enter email"
            onChange={(e) => setEmail(e.target.value)}
          />
          <Form.Text className="text-muted">
            We'll never share your email with anyone else.
          </Form.Text>
        </Form.Group>

        <Form.Group controlId="formBasicPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Password"
            onChange={(e) => setPassword(e.target.value)}
          />
        </Form.Group>
        {/* <Form.Group controlId="formBasicCheckbox">
          <Form.Check type="checkbox" label="Check me out" />
        </Form.Group> */}
        <Form.Group controlId="formBasicSubmit">
          <Button variant="primary" type="submit" className="full-width-btn">
            Login
          </Button>
        </Form.Group>
        <Form.Group controlId="formBasicNewAccount">
          <Button
            variant="secondary"
            className="full-width-btn"
            onClick={() => history.push("/register")}
          >
            Create New Account
          </Button>
        </Form.Group>
      </Form>
      {error ? (
        <Alert variant="danger" className="event-validation">
          {errorMessage}
        </Alert>
      ) : (
        <></>
      )}
    </Container>
  );
}
