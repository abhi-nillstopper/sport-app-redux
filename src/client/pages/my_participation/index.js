import React, { useState, useEffect } from "react";
import moment from "moment";
import api from "../../service/api";
import { Alert, Container } from "react-bootstrap";

export default function MyParticipation() {
  const [myEvents, setMyEvents] = useState([]);
  const [error, setError] = useState(false);
  const [errorMessage, setErrorMessage] = useState("");

  useEffect(() => {
    getMyParticipation();
  }, []);

  const getMyParticipation = async () => {
    try {
      const response = await api.get("/express/participation");
      setMyEvents(response.data);
    } catch (error) {
      setError(true);
      setErrorMessage(error.response.data.message);
    }
  };

  const isApproved = (approved) =>
    approved === true ? "Approved" : "Rejected";

  return (
    <Container className="myregistration-container">
      {error && (
        <Alert variant="danger" className="event-validation">
          {errorMessage}
        </Alert>
      )}
      <ul className="my-events">
        {myEvents.map((event) => (
          <li key={event._id}>
            <div>
              <strong>{event.eventTitle}</strong>
            </div>
              <span>
                Event Date: {moment(event.eventDate).format("DD,MMM,YY")}
              </span>
              <span>
                Event Price: â‚¹{parseFloat(event.eventPrice).toFixed(2)}
              </span>
              <span>User Email: {event.userEmail}</span>
              <span>
                Status:
                <span
                  className={
                    event.approved !== undefined
                      ? isApproved(event.approved)
                      : "Pending"
                  }
                >
                  {event.approved !== undefined
                    ? isApproved(event.approved)
                    : "Pending"}
                </span>
              </span>
          </li>
        ))}
      </ul>
    </Container>
  );
}
