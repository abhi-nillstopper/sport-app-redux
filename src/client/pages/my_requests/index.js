import React, { useState, useEffect } from "react";
import moment from "moment";
import api from "../../service/api";
import { Button, ButtonGroup, Alert, Container } from "react-bootstrap";
import "./my_requests.css";

export default function MyRequests() {
  const [myRequests, setMyRequests] = useState([]);
  const [error, setError] = useState(false);
  const [errorMessage, setErrorMessage] = useState("");
  const [success, setSuccess] = useState(false);
  const [successMessage, setSuccessMessage] = useState("");

  useEffect(() => {
    getMyRequests();
  }, []);

  const getMyRequests = async () => {
    try {
      const response = await api.get("/express/registration");
      setMyRequests(response.data);
    } catch (error) {
      setError(true);
      setErrorMessage(error.response.data.message);
    }
  };

  const isApproved = (approved) =>
    approved === true ? "Approved" : "Rejected";

  const handleApprove = async (eventId) => {
    try {
      await api.post(`/express/registration/${eventId}/approval`, {});
      successHandler(`Request Approved for ${eventId}`);
    } catch (error) {
      errorHandler(error);
    }
  };
  const handleReject = async (eventId) => {
    try {
      await api.post(`/express/registration/${eventId}/rejection`, {});
      successHandler(`Request Rejected for ${eventId}`);
    } catch (error) {
      errorHandler(error);
    }
  };

  const errorHandler = (message, logout = false) => {
    setError(true);
    setErrorMessage(message);
    setTimeout(() => {
      setError(false);
      // getMyRequests();
      //   logout ? logoutHandler() : dropDownHandler();
    }, 2000);
  };

  const successHandler = (message) => {
    setSuccess(true);
    setSuccessMessage(message);
    setTimeout(() => {
      setSuccess(false);
      getMyRequests();
    }, 2000);
  };

  return (
    <Container className="myregistration-container">
      {success && (
        <Alert variant="success" className="event-validation">
          {successMessage}
        </Alert>
      )}
      {error && (
        <Alert variant="danger" className="event-validation">
          {errorMessage}
        </Alert>
      )}
      <ul className="my-events">
        {myRequests.map((event) => (
          <li key={event._id}>
            <div>
              <strong>{event.eventTitle}</strong>
            </div>
              <span>
                Event Date: {moment(event.eventDate).format("DD,MMM,YY")}
              </span>
              <span>
                Event Price: â‚¹{parseFloat(event.eventPrice).toFixed(2)}
              </span>
              <span>User Email: {event.userEmail}</span>
              <span>
                Status:
                <span
                  className={
                    event.approved !== undefined
                      ? isApproved(event.approved)
                      : "Pending"
                  }
                >
                  {event.approved !== undefined
                    ? isApproved(event.approved)
                    : "Pending"}
                </span>
              </span>
            <ButtonGroup>
              <Button
                variant="primary"
                disabled={
                  event.approved === true || event.approved === false
                    ? true
                    : false
                }
                onClick={() => handleApprove(event._id)}
              >
                Approve
              </Button>
              <Button
                variant="danger"
                disabled={
                  event.approved === true || event.approved === false
                    ? true
                    : false
                }
                onClick={() => handleReject(event._id)}
              >
                Reject
              </Button>
            </ButtonGroup>
          </li>
        ))}
      </ul>
    </Container>
  );
}
