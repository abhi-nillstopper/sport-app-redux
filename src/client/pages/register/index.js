import React, { useState, useContext } from "react";
import { Form, Button, Container, Alert } from "react-bootstrap";
import api from "../../service/api";
import { UserContext } from "../../user-context";
import "./register.css"

export default function Login({ history }) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [fieldEmpty, setFieldEmpty] = useState(false);
  const [errorMessage, setErrorMessage] = useState("");
  const { setIsLoggedIn } = useContext(UserContext);

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (
      email !== "" &&
      password !== "" &&
      firstName !== "" &&
      lastName !== ""
    ) {
      const response = await api.post("/express/user/register", {
        email,
        password,
        firstName,
        lastName,
      });
      const user_id = response.data.user_id || false;
      const user = response.data.user || false;

      if (user_id && user) {
        localStorage.setItem("user_id", user_id);
        localStorage.setItem("user", user);
        setIsLoggedIn(true);
        history.push("/");
      } else {
        const { message } = response.data;
        errorHandler(message);
      }
    } else {
      errorHandler("You need to fill all the inputs");
    }
  };

  const errorHandler = (message) => {
    setFieldEmpty(true);
    setErrorMessage(message);
    setTimeout(() => setFieldEmpty(false), 2000);
  };

  return (
    <Container className="register-container">
      <h2>Register:</h2>
      <p>
        Please <strong>register</strong> for your account
      </p>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId="formBasicFirstName">
          <Form.Label>First Name</Form.Label>
          <Form.Control
            type="text"
            name="firstName"
            placeholder="Enter firstName"
            onChange={(e) => setFirstName(e.target.value)}
          />
        </Form.Group>

        <Form.Group controlId="formBasicLastName">
          <Form.Label>Last Name</Form.Label>
          <Form.Control
            type="text"
            name="lastName"
            placeholder="Enter lastName"
            onChange={(e) => setLastName(e.target.value)}
          />
        </Form.Group>

        <Form.Group controlId="formBasicEmail">
          <Form.Label>Email address</Form.Label>
          <Form.Control
            type="email"
            name="email"
            placeholder="Enter email"
            onChange={(e) => setEmail(e.target.value)}
          />
          <Form.Text className="text-muted">
            We'll never share your email with anyone else.
          </Form.Text>
        </Form.Group>

        <Form.Group controlId="formBasicPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Password"
            onChange={(e) => setPassword(e.target.value)}
          />
        </Form.Group>
        {/* <Form.Group controlId="formBasicCheckbox">
          <Form.Check type="checkbox" label="Check me out" />
        </Form.Group> */}
        <Form.Group controlId="formBasicSubmit">
          <Button variant="primary" type="submit" className="full-width-btn">
            Submit
          </Button>
        </Form.Group>
        <Form.Group controlId="formBasicLogin">
          <Button
            variant="secondary"
            onClick={() => history.push("/login")}
            className="full-width-btn"
          >
            Login
          </Button>
        </Form.Group>
      </Form>
      {fieldEmpty ? (
        <Alert variant="danger" className="event-validation">
          {errorMessage}
        </Alert>
      ) : (
        <></>
      )}
    </Container>
  );
}
