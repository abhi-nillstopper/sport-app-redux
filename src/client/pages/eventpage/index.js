import React from "react";
import { withRouter } from "react-router-dom";
import {
  Form,
  Button,
  Container,
  Image,
  Alert,
  Dropdown,
  SplitButton,
} from "react-bootstrap";
import api from "../../service/api";
import autoBind from "react-autobind";
import {ReactComponent as CameraIcon} from "../../assets/camera.svg";
import "./events.css";

//Create Event
class EventPage extends React.Component {
  user_id;
  user;
  dropDownItems;
  constructor(props) {
    super(props);
    this.user_id = localStorage.getItem("user_id");
    this.user = localStorage.getItem("user");
    this.dropDownItems = [
      { key: "running", itemName: "Running" },
      { key: "cycling", itemName: "Cycling" },
      { key: "swimming", itemName: "Swimming" },
    ];
    this.state = {
      title: "",
      description: "",
      price: "",
      thumbnail: null,
      date: "",
      sport: "",
      preview: null,
      fieldEmpty: false,
      success: false,
    };

    // this.handleOnChange = this.handleOnChange.bind(this);
    autoBind(this);
  }

  // shouldComponentUpdate(nextProps, nextState) {
  //   console.log("user", this.user)
  //   if (!this.user) {
  //     this.props.history.push("/login");
  //     return false;
  //   }
  //   return true;
  // }

  componentDidMount() {
    if (!this.user) {
      this.props.history.push("/login");
    } else {
      document.querySelector(
        `div[name="sport-dropdown"]`
      ).children[0].disabled = true;
    }
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    let { thumbnail } = this.state;
    if (prevState.thumbnail !== thumbnail) {
      this.setState({
        preview: thumbnail ? URL.createObjectURL(thumbnail) : null,
      });
    }
  }

  async submitHandler(e) {
    e.preventDefault();
    let {
      date,
      description,
      price,
      sport,
      thumbnail,
      title,
      // success,
    } = this.state;
    const eventData = new FormData();
    eventData.append("title", title);
    eventData.append("description", description);
    eventData.append("price", price);
    eventData.append("thumbnail", thumbnail);
    eventData.append("date", date);
    eventData.append("sport", sport);

    try {
      if (
        title !== "" &&
        description !== "" &&
        price !== "" &&
        thumbnail !== null &&
        date !== "" &&
        sport !== ""
      ) {
        await api.post("/express/event", eventData);
        this.setState({ success: true });
        setTimeout(() => {
          this.setState({ success: false });
          this.props.history.push("/");
        }, 2000);
      } else {
        this.setState({ fieldEmpty: true });
        setTimeout(() => this.setState({ fieldEmpty: false }), 2000);
      }
    } catch (err) {
      console.log(err);
      Promise.reject(err);
    }
    return;
  }

  handleOnChange(e) {
    if (!e.target.files) {
      this.setState({ [e.target.name]: e.target.value });
    } else {
      this.setState({ [e.target.name]: e.target.files[0] });
    }
    // setTimeout(()=> console.log(this.state),1000)
  }

  onSelectSport(eventKey, evt) {
    this.setState({ sport: eventKey });
    document.querySelector(`div[name="sport-dropdown"]`).children[0].innerHTML =
      evt.target.innerHTML;
  }

  render() {
    const { history } = this.props;
    let {
      date,
      description,
      price,
      thumbnail,
      title,
      preview,
      fieldEmpty,
      success,
    } = this.state;
    return (
      <Container className="new-event-container">
        <h2>Create your event</h2>
        <Form onSubmit={this.submitHandler}>
          <Form.Group controlId="formBasicThumbnail">
            <Form.Label>Upload Image:</Form.Label>
            <Form.Label
              id="thumbnail"
              style={{ backgroundImage: `url(${preview})` }}
              className={thumbnail ? "has-thumbnail" : ""}
            >
              <Form.File name="thumbnail" onChange={this.handleOnChange} />
              <CameraIcon/>
            </Form.Label>
          </Form.Group>
          <Form.Group controlId="formBasicTitle">
            <Form.Label>Title:</Form.Label>
            <Form.Control
              type="text"
              name="title"
              value={title}
              placeholder="Enter Title"
              onChange={this.handleOnChange}
            />
          </Form.Group>
          <Form.Group controlId="formBasicDescription">
            <Form.Label>Description:</Form.Label>
            <Form.Control
              type="text"
              name="description"
              value={description}
              placeholder="Enter Description"
              onChange={this.handleOnChange}
            />
          </Form.Group>
          <Form.Group controlId="formBasicPrice">
            <Form.Label>Price:</Form.Label>
            <Form.Control
              type="number"
              name="price"
              value={price}
              placeholder="Enter Price in â‚¹"
              onChange={this.handleOnChange}
            />
          </Form.Group>

          <Form.Group controlId="formBasicDate">
            <Form.Label>Date:</Form.Label>
            <Form.Control
              type="date"
              name="date"
              value={date}
              placeholder="Enter Date"
              onChange={this.handleOnChange}
            />
          </Form.Group>
          <Form.Group controlId="formBasicSport">
            {/* <Form.Label>Sport:</Form.Label> */}
            <SplitButton
              variant={"secondary"}
              name="sport-dropdown"
              title={"Sport"}
            >
              <Dropdown.Item eventKey="1" disabled>
                Select Sport
              </Dropdown.Item>
              <Dropdown.Divider />
              {this.dropDownItems.map((drpdwn, idx) => {
                return (
                  <Dropdown.Item
                    key={idx}
                    eventKey={drpdwn.key}
                    onSelect={this.onSelectSport}
                  >
                    {drpdwn.itemName}
                  </Dropdown.Item>
                );
              })}
            </SplitButton>
          </Form.Group>
          <Form.Group controlId="formBasicSubmit">
            <Button variant="primary" type="submit" className="full-width-btn">
              Create Event
            </Button>
          </Form.Group>
          <Form.Group controlId="formBasicDashboard">
            <Button
              variant="danger"
              onClick={() => history.push("/")}
              title="Dashboard"
              className="full-width-btn"
            >
              Cancel
            </Button>
          </Form.Group>
        </Form>
        {fieldEmpty ? (
          <Alert variant="danger" className="event-validation">
            missing required fields
          </Alert>
        ) : (
          <></>
        )}
        {success ? (
          <Alert variant="success" className="event-validation">
            Event created
          </Alert>
        ) : (
          <></>
        )}
      </Container>
    );
  }
}

export default withRouter(EventPage);
