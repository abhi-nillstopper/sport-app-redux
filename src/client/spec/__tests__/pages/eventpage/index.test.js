import React from "react";
import EventPage from "../../../../pages/eventpage";
import { PureEventPage } from "../../../../pages/eventpage";
import { mount, shallow } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { defaultProps, fileAsInput } from "./defaultProps";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { withoutHooks } from "jest-react-hooks-shallow";

const mockStore = configureMockStore();
const store = mockStore({
  event: { success: true },
  /* here you can create a mock object for the reducer */
});

describe("Event page snapshot", () => {
  let spy = spyConsole();

  //   beforeEach(() => {
  //     defaultProps.createEvents = jest.fn();
  //     defaultProps.changeSuccess = jest.fn();
  //     defaultProps.success = false;
  //   });

  it("renders without crashing", () => {
    withoutHooks(() => {
      const tree = renderer.create(<PureEventPage {...defaultProps} />);
      // console.log(tree);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});

describe("Event page snapshot", () => {
  let spy = spyConsole();
  let mockEvent;
  let wrapper;
  beforeAll(() => {
    mockEvent = { preventDefault: () => {}, target: { files: null } };
  });

  beforeEach(() => {
    defaultProps.createEvents = jest.fn();
    defaultProps.changeSuccess = jest.fn();
    defaultProps.success = false;
  });
  it("simulate submit", () => {
    withoutHooks(() => {
      wrapper = mount(<PureEventPage {...defaultProps} />);
      const spyOnEvent = jest.spyOn(wrapper.instance(), "submitHandler");
      wrapper.instance().submitHandler(mockEvent);
      //   wrapper.find('[type="submit"]').first().simulate("submit", mockEvent);
      expect(spyOnEvent).toBeCalledWith(mockEvent);
    });
  });

  it.skip("simulate onChange file", () => {
    withoutHooks(() => {
      wrapper = mount(<PureEventPage {...defaultProps} />);
      const spyOnEvent = jest.spyOn(wrapper.instance(), "handleOnChange");
      // let fileInput = wrapper.find("input").find('[type="file"]');
      // fileInput.getDOMNode().files = [fileAsInput];
      fileInput.simulate("change", mockEvent);
      expect(spyOnEvent).toBeCalledWith(mockEvent);
    });
  });

  it("simulate onChange value", () => {
    withoutHooks(() => {
      wrapper = mount(<PureEventPage {...defaultProps} />);
      const spyOnEvent = jest.spyOn(wrapper.instance(), "handleOnChange");
      let input = wrapper.find("input").find('[type="text"]').first();
      // fileInput.getDOMNode().files = [fileAsInput];
      input.instance().value = "abhi";
      mockEvent = { ...mockEvent, target: { value: "abhi" } };
      wrapper.instance().handleOnChange(mockEvent);
      wrapper.update();
      input.simulate("change", mockEvent);
      expect(spyOnEvent).toBeCalled();
      expect(spyOnEvent).lastCalledWith(mockEvent);

    });
  });
});

function spyConsole() {
  // https://github.com/facebook/react/issues/7047
  let spy = {};

  beforeAll(() => {
    spy.console = jest.spyOn(console, "error").mockImplementation(() => {});
  });

  afterAll(() => {
    spy.console.mockRestore();
  });

  return spy;
}
