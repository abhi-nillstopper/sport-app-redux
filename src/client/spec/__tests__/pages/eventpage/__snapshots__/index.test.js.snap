// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event page snapshot renders without crashing 1`] = `
<div
  className="new-event-container container"
>
  <h2>
    Create your event
  </h2>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicThumbnail"
      >
        Upload Image:
      </label>
      <label
        className="form-label"
        htmlFor="formBasicThumbnail"
        id="thumbnail"
        style={
          Object {
            "backgroundImage": "url(null)",
            "backgroundSize": "cover",
          }
        }
      >
        <div
          className="form-file"
        >
          <input
            className="form-control-file"
            disabled={false}
            id="formBasicThumbnail"
            name="thumbnail"
            onChange={[Function]}
            type="file"
          />
        </div>
        <span />
      </label>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicTitle"
      >
        Title:
      </label>
      <input
        className="form-control"
        id="formBasicTitle"
        name="title"
        onChange={[Function]}
        placeholder="Enter Title"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicDescription"
      >
        Description:
      </label>
      <input
        className="form-control"
        id="formBasicDescription"
        name="description"
        onChange={[Function]}
        placeholder="Enter Description"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicPrice"
      >
        Price:
      </label>
      <input
        className="form-control"
        id="formBasicPrice"
        name="price"
        onChange={[Function]}
        placeholder="Enter Price in â‚¹"
        type="number"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="formBasicDate"
      >
        Date:
      </label>
      <input
        className="form-control"
        id="formBasicDate"
        name="date"
        onChange={[Function]}
        placeholder="Enter Date"
        type="date"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <div
        className="dropdown btn-group"
        name="sport-dropdown"
        role="group"
      >
        <button
          className="btn btn-secondary"
          disabled={false}
          type="button"
        >
          Sport
        </button>
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle dropdown-toggle-split btn btn-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle dropdown
          </span>
        </button>
      </div>
    </div>
    <div
      className="form-group"
    >
      <button
        className="full-width-btn btn btn-primary"
        disabled={false}
        type="submit"
      >
        Create Event
      </button>
    </div>
    <div
      className="form-group"
    >
      <button
        className="full-width-btn btn btn-danger"
        disabled={false}
        onClick={[Function]}
        title="Dashboard"
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
